app/src/main/cpp/skyline/input/touch.cpp

```diff
@@ -26,17 +26,23 @@ namespace skyline::input {
         entry.localTimestamp = lastEntry.localTimestamp + 1;
         entry.touchCount = points.size();
 
-        for (size_t i{}; i < points.size(); i++) {
+        for (size_t i{}; i < points.size() && i < MaxTouchPoints; i++) {
```

Comment:
Is it fine if points.size() is greater than 16? Cause if it is then touchcount will be larger than the number of populated points

Reply:
It's not fine if it's bigger since you overflow the array. Although it's really hard to put 16 fingers on a touch screen even if you got 3 hands.

I can add a guarantee from the UI side so this can never happen here

Reply:
Ah okay probably best to just do it at start of function, just do points = points.subspan(0, STD::min(points.size(),  entry.data.size()))

Oh also can you make it take span directly rather than const ref, not sure how that snook past review

Reply:
@bylaws It's older code, I'm not sure if we were even using spans much back then.